src/frontend/apps/impress/src/features/docs/doc-editor/components/AIButton.tsx

```diff
@@ -321,19 +265,45 @@ const AIMenuItemTranslate = ({
 };
 
 interface AIMenuItemProps {
-  handleAIAction: () => Promise<void>;
+  requestAI: (markdown: string) => Promise<string>;
   isPending: boolean;
   icon?: ReactNode;
 }
 
 const AIMenuItem = ({
-  handleAIAction,
+  requestAI,
   isPending,
   children,
   icon,
 }: PropsWithChildren<AIMenuItemProps>) => {
   const Components = useComponentsContext();
 
+  const editor = useBlockNoteEditor();
+  const handleAIError = useHandleAIError();
+
+  const handleAIAction = async () => {
+    const selectedBlocks = editor.getSelection()?.blocks;
+
+    if (!selectedBlocks || selectedBlocks.length === 0) {
+      return;
+    }
+
+    const markdown = await editor.blocksToMarkdownLossy(selectedBlocks);
+
+    try {
+      const responseAI = await requestAI(markdown);
+
+      if (!responseAI) {
+        return;
+      }
```

Comment:
I'm not a big fan of this generic way. I would explain more because Depending on the type, it can only be a string and never undefined or null. So I would try a string

```
if (responseAI === '' ) {
    return;
}
```